src/ui/public/route_based_notifier/index.js

```diff
@@ -0,0 +1,47 @@
+import { includes, mapValues } from 'lodash';
+import Notifier from 'ui/notify/notifier';
+
+/*
+ * Caches notification attempts so each one is only actually sent to the
+ * notifier service once per route.
+ */
+export default function routeBasedNotifierProvider($rootScope) {
+  const notifier = new Notifier();
+
+  let notifications = {
+    warnings: []
```

Comment:
Any reason this only supports warnings? Is it just because that's all you needed to use right now?